x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -251,6 +253,39 @@ export class ActionExecutor {
 
         event.event = event.event || {};
 
+        // start gen_ai extension
+        // add event.kibana.action.execution.gen_ai to event log when GenerativeAi Connector is executed
+        if (result.status === 'ok' && actionTypeId === '.gen-ai') {
+          const data = result.data as unknown as {
+            usage: { prompt_tokens: number; completion_tokens: number; total_tokens: number };
+          };
+          event.kibana = event.kibana || {};
+          event.kibana.action = event.kibana.action || {};
+          event.kibana = {
+            ...event.kibana,
+            action: {
+              ...event.kibana.action,
+              execution: {
+                ...event.kibana.action.execution,
+                gen_ai: {
+                  usage: {
+                    total_tokens: data.usage.total_tokens,
```

Comment:
I added a comment to the commit, but it doesn't show here, so repeating :-)

If `data.usage` isn't set, I believe the field references will error will a null dereference.  I think we can just change `usage.X` to `usage?.X`

Reply:
:shipit: